INSTALL.rst

```diff
@@ -0,0 +1,363 @@
+    2.7 only, on Linux and mac os X only)
+
+Optionally, you can also install a number of packages to enables better user
+interface toolkit. See :ref:`what-is-a-backend` for more details on the
+optional Matplotlib backends and the capabilities they provide.
+
+  * :term:`tk` (>= 8.3, != 8.6.0 or 8.6.1): for the TkAgg backend;
+  * `PyQt4 <https://pypi.python.org/pypi/PyQt4>`_ (>= 4.4): for the Qt4Agg backend;
+  * `PyQt5 <https://pypi.python.org/pypi/PyQt5>`_: for the Qt5Agg backend;
+  * :term:`pygtk` (>= 2.4): for the GTK and the GTKAgg backend;
+  * :term:`wxpython` (>= 2.8 or later): for the WX or WXAgg backend;
+  * `pycairo <https://pypi.python.org/pypi/pycairo>`_;
+  * `PySide <https://pypi.python.org/pypi/PySide>`_;
+  * `PyGTK <https://pypi.python.org/pypi/PyGTK>`_;
+  * `Tornado <https://pypi.python.org/pypi/tornado>`_: for the WebAgg backend.
+
+For better support of animation output format and image file format, LaTeX,
+etc you can
+install the following:
+
+  * `ffmpeg <https://www.ffmpeg.org/>`__/`avconv
+    <https://libav.org/avconv.html>`__ or `mencoder
+    <http://www.mplayerhq.hu/design7/news.html>`__ (for saving movies);
+  * `ImageMagick <http://www.imagemagick.org/script/index.php>`__ (for saving
+    animated gifs);
+  * `Pillow <http://python-pillow.org/>`__ (for a larger selection of image
+    file format: JPEG, BMP, and TIFF image files);
+  * `Latex <https://miktex.org/>`_ and `GhostScript <https://ghostscript.com/download/>`_ 
+    (for rendering text with LaTeX);
+
+.. note::
+
+   Matplotlib depends on a large number of non-python libraries. 
+   `pkg-config <https://www.freedesktop.org/wiki/Software/pkg-config/>`__
+   can be used to find required non-python libraries and thus make install go
+   more smoothly if the libraries and headers are not in the expected
+   locations.
+
+.. note::
+
+  The following libraries are shipped with Matplotlib:
+
+    - `Agg`: the antigrain C++ rendering engine;
+    - `qhull`: to compute Delaunay triangulation;
+    - `ttconv`: a true type font utility.
+
+.. _build_linux:
+
+Building on Linux
+-----------------
+
+It is easiest to use your system package manager to install the dependencies.
+
+If you are on Debian/Ubuntu, you can get all the dependencies
+required to build Matplotlib with::
+
+   sudo apt-get build-dep python-matplotlib
+
+If you are on Fedora/RedHat, you can get all the dependencies required
+to build Matplotlib by first installing ``yum-builddep`` and then
+running::
+
+   su -c "yum-builddep python-matplotlib"
+
+This does not build Matplotlib, but it does get and install the
+build dependencies, which will make building from source easier.
+
+
+.. _build_osx:
+
+Building on OSX
+---------------
+
+The build situation on OSX is complicated by the various places one
+can get the libpng and freetype requirements (darwinports, fink,
+/usr/X11R6) and the different architectures (e.g., x86, ppc, universal) and
```

Comment:
"and" -> ","
plus a second comma before the last "and" if you like oxford commas.